internal/ingest/ingest.go

```diff
@@ -71,7 +71,7 @@ func NewLegIngester(ctx context.Context, cfg config.Ingest, h host.Host,
 	idxr *indexer.Engine, reg *registry.Registry, ds datastore.Batching) (LegIngester, error) {
 
 	lsys := mkLinkSystem(ds, reg)
-	lms, err := legs.NewMultiSubscriber(ctx, h, ds, lsys, cfg.PubSubTopic)
+	lms, err := legs.NewMultiSubscriber(ctx, h, ds, lsys, cfg.PubSubTopic, nil)
```

Comment:
@gammazero @willscott I'd love your input on whether there are any plans to utilise selectors in storetheindex. For now setting this to `nil` so that the work on dependency restructuring can carry on in reference provider.

Reply:
Yes. this interface change came from an ask we got in integration:
* The primary legs sync should only sync the advertisments. Each list of CIDs getting pulled over backed by a car should be it's own graphsync session. Otherwise, you have a risk of memory exhaustion on the server, since one transaction will open and retain references to all deal files since graphsync holds onto previously transferred blocks in order to manage deduplication within a session.

Reply:
Selector matcher with `Has_field` condition is not implemented in IPLD.

Can you think of a way to work around it here (other than implementing the selector condition in IPLD)?

Reply:
I think you want a recursive selector but instead of a 'explore all' in the recursion you want an ['explore fields'](https://github.com/ipld/go-ipld-prime/blob/master/traversal/selector/exploreFields.go#L20)

Reply:
sigh... thank you

Reply:
@willscott  would you mind taking a look at my selector to check it's the right one? 🍻

Reply:
that looks like the right thing to me 👍